src/main/java/data_access/PodcastDataAccessObject.java

```diff
@@ -0,0 +1,124 @@
+package data_access;
+
+import entities.*;
+import java.io.*;
+import java.net.URISyntaxException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class PodcastDataAccessObject implements PodcastDataAccess {
+
+    private final Map<UUID, Podcast> podcastMap;
+    private final EpisodeDataAccess episodeDAO;
+
+    public PodcastDataAccessObject(EpisodeDataAccess episodeDAO) {
+        podcastMap = new HashMap<>();
+        this.episodeDAO = episodeDAO;
+        loadPodcasts();
+    }
+
+    @Override
+    public List<MediaItem> getEpisodesForPodcast(UUID podcastId) {
+        Podcast podcast = podcastMap.get(podcastId);
+        return podcast.getItems();
+    }
+
+    @Override
+    public Podcast getPodcastById(UUID id) {
+        return podcastMap.get(id);
+    }
+
+    @Override
+    public boolean savePodcast(Podcast podcast) {
+        podcastMap.put(podcast.getId(), podcast);
+        return save();
+    }
+
+    private void loadPodcasts() {
+        File podcastsCSV;
+        try {
+            podcastsCSV = new File(this.getClass().getResource("/podcasts.csv").toURI());
+        } catch (URISyntaxException e) {
+            System.out.println("Error reading podcasts.csv.");
+            e.printStackTrace();
+            return;
+        }
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(podcastsCSV));
+            reader.readLine();  // read the header row
+            String row;
+            while ((row = reader.readLine()) != null) {
+                String[] col = row.split(",");
```

Comment:
This is where the wrong assumption is (see other comment). If the name has a comma, it will split the name is different cols.

Reply:
I'll create an additional branch to address this issue, just so we can work with the debugging features within Github.